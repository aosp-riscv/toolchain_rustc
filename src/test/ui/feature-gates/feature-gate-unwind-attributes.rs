// compile-flags: -C no-prepopulate-passes -Cpasses=name-anon-globals

#![crate_type = "lib"]

extern {
// CHECK: Function Attrs: nounwind
// CHECK-NEXT: declare void @extern_fn
    fn extern_fn();
// CHECK-NOT: Function Attrs: nounwind
// CHECK: declare void @unwinding_extern_fn
<<<<<<< HEAD   (086005 Importing rustc-1.38.0)
    #[unwind(allowed)] //~ ERROR `#[unwind]` is experimental
=======
    #[unwind(allowed)] //~ ERROR the `#[unwind]` attribute is an experimental feature
>>>>>>> BRANCH (8cd2c9 Importing rustc-1.39.0)
    fn unwinding_extern_fn();
}

pub unsafe fn force_declare() {
    extern_fn();
    unwinding_extern_fn();
}
