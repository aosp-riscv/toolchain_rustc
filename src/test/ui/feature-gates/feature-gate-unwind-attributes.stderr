<<<<<<< HEAD   (086005 Importing rustc-1.38.0)
error[E0658]: `#[unwind]` is experimental
=======
error[E0658]: the `#[unwind]` attribute is an experimental feature
>>>>>>> BRANCH (8cd2c9 Importing rustc-1.39.0)
  --> $DIR/feature-gate-unwind-attributes.rs:11:5
   |
LL |     #[unwind(allowed)]
   |     ^^^^^^^^^^^^^^^^^^
   |
   = note: for more information, see https://github.com/rust-lang/rust/issues/58760
   = help: add `#![feature(unwind_attributes)]` to the crate attributes to enable

error: aborting due to previous error

For more information about this error, try `rustc --explain E0658`.
